syntax = "proto3";

package mes_grpc;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service MessengerServer {
    rpc SendMessage(MessageRequest) returns (MessageResponse);
    rpc ReadMessages(google.protobuf.Empty) returns (stream Message);
}

message MessageRequest {
    string author = 1;
    string text = 2;
}

message MessageResponse {
    google.protobuf.Timestamp sendTime = 1;
}

message Message {
    string author = 1;
    string text = 2;
    google.protobuf.Timestamp sendTime = 3;
}
